File: app/src/main/java/com/karthek/android/s/ainfo/MainActivity.java
package com.karthek.android.s.ainfo;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;

import com.karthek.android.s.ainfo.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {

	private ActivityMainBinding binding;


	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		binding = ActivityMainBinding.inflate(getLayoutInflater());
		View view = binding.getRoot();
		setContentView(view);
		getSupportFragmentManager().beginTransaction()
				.setReorderingAllowed(true)
				.add(R.id.fragment, HFragment.newInstance())
				.commit();
	}
}

File: app/src/main/java/com/karthek/android/s/ainfo/MainActivity.kt
package com.karthek.android.s.ainfo

import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.activity.enableEdgeToEdge
import androidx.compose.foundation.LocalIndication
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.material3.Surface
import androidx.compose.runtime.Composable
import androidx.compose.runtime.CompositionLocalProvider
import androidx.compose.ui.Modifier
import com.karthek.android.s.ainfo.ui.screens.MainScreen
import com.karthek.android.s.ainfo.ui.theme.AppTheme
import com.karthek.android.s.helper.ui.components.ScaleIndication

class MainActivity : ComponentActivity() {
	override fun onCreate(savedInstanceState: Bundle?) {
		super.onCreate(savedInstanceState)
		enableEdgeToEdge()
		setContent { ScreenContent() }
	}

	@Composable
	fun ScreenContent() {
		AppTheme {
			Surface(modifier = Modifier.fillMaxSize()) {
				CompositionLocalProvider(LocalIndication provides ScaleIndication) {
					MainScreen()
				}
			}
		}
	}
}